### ISSUE TITLE
Mentions in comments make two "new comment" notifications possible

### ISSUE THREAD

**Issue by Bonsaiheldin** [diaspora/diaspora#7405]
The new **mentions in comments** feature got yet another bug (besides being untranslated). One gets at least two notifications for multiple comments, if one of them contains a mention. Diaspora should check if one of them was already created as a comment containing a mention is still a comment.

[![](https://picload.org/infonail/rlwpddcw/screenshotat2017-03-28224146.jpg)](https://picload.org/image/rlwpddcw/screenshotat2017-03-28224146.png)

At the same time, this is also the very first GitHub issue i ever opened, anywhere! Yay! :1st_place_medal:

**Comment by cmrd-senya** [comment]
Do you mean you get at least two notifications per a comment? Each comment is notified separately, so for a comment with a mention you get "mentioned in comment notification" while if there is no mention and you're a participant you'll get "also commented" or "comment on your post" notifications. So if there is two new comments in the thread you'll get two notifications, one for each comment. It is how it is expected to behave.

**Comment by Bonsaiheldin** [comment]
Okay... Yes, i get a notification for a comment and a notification if it contains a mention - not at the same time, only when there are two comments.

I understand how it can be meant to be that way. But to me it's just two notifications for comments from the same person. I think it's unnecessary as both notifications notify me about a new comment. But oh well...

**Comment by cmrd-senya** [comment]
That's not a bug, but perhaps we can consider this as a feature request. But in this case I think it needs more discussion.

**Comment by SuperTux88** [comment]
Yes, in the current notification-system this isn't a bug, because it works as designed (because that are different types of notifications).

But we probably need to rethink and improve the notification-system, to make it possible to group different notifications. And that needs more discussion to decide how notifications should work in the future (that's why I added the "needs discourse" label).

**Comment by Flaburgan** [comment]
Linked: https://github.com/diaspora/diaspora/issues/6623

**Comment by SuperTux88** [comment]
Some additional information: Even if there are multiple notifications, if you visit the post to read the comments, all notifications are marked as read :)

**Comment by Flaburgan** [comment]
Also linked to #4630
